util/logtools/make_timeline.ts

```diff
@@ -148,42 +136,44 @@ timelineParse.parser.addArgument(['--include_targetable', '-it'], {
   help: 'Set this flag to include "34" log lines when making the timeline',
 });
 
-const args: Partial<ExtendedArgsNamespace> = new ExtendedArgsNamespace({});
-timelineParse.parser.parseArgs(undefined, args);
-
 const printHelpAndExit = (errString: string): void => {
   console.error(errString);
   timelineParse.parser.printHelp();
   process.exit(-1);
 };
 
-const hasFile = typeof args.file === 'string' && args?.file !== '';
-const hasReport = typeof args.report_id === 'string' && args?.report_id !== '';
-
-// TODO: Revisit this logic when we re-add FFLogs support.
-if (hasFile && hasReport || !hasFile && !hasReport)
-  printHelpAndExit('Error: Must specify exactly one of -f or -r\n');
-if (hasFile && !args.file?.includes('.log'))
-  printHelpAndExit('Error: Must specify an FFXIV ACT log file, as log.log\n');
-if (hasReport && typeof args.report_id !== 'string')
-  printHelpAndExit('Error: Must specify a report ID.');
-let numExclusiveArgs = 0;
-const exclusiveArgs = ['search_fights', 'search_zones', 'report_fight'] as const;
-for (const opt of exclusiveArgs) {
-  if (args[opt] !== null && args[opt] !== undefined)
-    numExclusiveArgs++;
-}
-if (numExclusiveArgs !== 1)
-  printHelpAndExit('Error: Must specify exactly one of -lf, -lz, or -rf\n');
-if (
-  typeof args.report_id === 'string' &&
-  (typeof args.report_fight !== 'number' || args.report_fight < 0)
-)
-  printHelpAndExit('Error: Must specify a report fight index of 0 or greater');
-if (args.fight_regex === '-1')
-  printHelpAndExit('Error: Timeline generation does not currently support -fr\n');
-if (args.zone_regex === '-1')
-  printHelpAndExit('Error: Timeline generation does not currently support -zr\n');
+const validateArgs = (args: ExtendedArgs): void => {
```

Comment:
Oh, of course, sweeping all the validation into a function makes a lot more sense than keeping it in-line.